app/src/handle-request.lisp

```diff
@@ -57,142 +57,143 @@ The mapping vector V specifies that the qubit as specified in the program V[i] h
          (type (gethash "type" js))
          (gate-noise (gethash "gate-noise" js))
          (measurement-noise (gethash "measurement-noise" js)))
-    (qvm:with-random-state ((get-random-state (gethash "rng-seed" js)))
-      ;; Basic logging
-      (format-log "Got ~S request from API key/User ID: ~S / ~S" type api-key user-id)
-      ;; Dispatch
-      (ecase (keywordify type)
-        ;; For simple tests.
-        ((:ping)
-         (handle-ping))
-
-        ;; Get the version of everything.
-        ((:version)
-         (handle-version))
-
-        ;; Get misc info about the simulation mode, shared memory, etc.
-        ((:info)
-         (handle-info))
-
-        ;; Multishot experiments.
-        ((:multishot)
-         (check-required-fields js "addresses" "trials")
-         (check-for-quil-instrs-field js)
-         (let* ((addresses (gethash "addresses" js))
-                (num-trials (gethash "trials" js))
-                (isns (get-quil-instrs-field js))
-                (quil (let ((quil::*allow-unresolved-applications* t))
-                        (process-quil (safely-parse-quil-string isns))))
-                (num-qubits (cl-quil:qubits-needed quil))
-                (results (perform-multishot *simulation-method* quil num-qubits addresses num-trials
-                                            :gate-noise gate-noise
-                                            :measurement-noise measurement-noise)))
-           (check-type results hash-table)
-           (with-output-to-string (s)
-             (yason:encode results s))))
-
-        ;; Multishot with final measure.
-        ((:multishot-measure)
-         (check-required-fields js "qubits" "trials")
-         (check-for-quil-instrs-field js)
-         (let* ((quil:*allow-unresolved-applications* t)
-                (qubits (gethash "qubits" js))
-                (num-trials (gethash "trials" js)))
-           (multiple-value-bind (quil relabeling)
-               (process-quil (safely-parse-quil-string
-                              (get-quil-instrs-field js)))
-             (let* ((num-qubits (cl-quil:qubits-needed quil))
-                    (results (perform-multishot-measure
-                              *simulation-method*
-                              quil
-                              num-qubits
-                              qubits
-                              num-trials
-                              relabeling)))
+    (prog1
```

Comment:
This `prog1` is unsightly since the first form is yuge. Alternatively, we could move the gc to the start of the request, rather than the end.